
# Week 2 Assignment: Deriving Insight via Descriptive Statistics

### 1. Inspect the wages.csv dataset 

we will inspect the wages.csv dataset to provide a detailed description including the variable names, descriptions, and data types for each variable.

For this load the data set

In [1]:
import pandas as pd

In [2]:
# read the file and save data in a data frame named 'df' and veiw it
df = pd.read_csv('C:/Users/balar/Downloads/G/G/assign_wk2/assign_wk2/wages.csv')
df

Surname       Name  Age  Gender                  Ethnicity Start_date  \
0          Bold   Caroline   63  Female                      White   7/2/2012   
1        Zamora   Jennifer   38  Female                      White  4/10/2010   
2      Houlihan      Debra   51  Female                      White   5/5/2014   
3      Bramante      Elisa   34  Female  Black or African American   1/5/2009   
4    Del Bosque      Keyla   38  Female  Black or African American   1/9/2012   
..          ...        ...  ...     ...                        ...        ...   
169    Osturnka      Adeel   41    Male                      White  9/30/2013   
170    Punjabhi      Louis   56    Male                      White   1/6/2014   
171      Cockel      James   40    Male                      White   7/8/2013   
172      Gordon      David   38    Male                      White   7/2/2012   
173       Knapp  Bradley J   40    Male  Black or African American  2/17/2014   

               Department                 Position       Salary  
0        Executive Office          President & CEO  $166,400.00  
1                   IT/IS                      CIO  $135,200.00  
2                   Sales        Director of Sales  $124,800.00  
3              Production   Director of Operations  $124,800.00  
4    Software Engineering        Software Engineer  $118,809.60  
..                    ...                      ...          ...  
169            Production  Production Technician I   $33,280.00  
170            Production  Production Technician I   $33,280.00  
171            Production  Production Technician I   $31,200.00  
172            Production  Production Technician I   $31,200.00  
173            Production  Production Technician I   $29,120.00  

[174 rows x 9 columns]

The dataset consists of several columns such as Surname, Name, Age, Gender, Ethnicity, Start_date, Department, Position, and Salary.
Each row represents an employee of the company.

We can use info function to know the data type of each coloumns in our data set

In [3]:
# info function implementation to our data set
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 174 entries, 0 to 173
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Surname     174 non-null    object
 1   Name        174 non-null    object
 2   Age         174 non-null    int64 
 3   Gender      174 non-null    object
 4   Ethnicity   174 non-null    object
 5   Start_date  174 non-null    object
 6   Department  174 non-null    object
 7   Position    174 non-null    object
 8   Salary      174 non-null    object
dtypes: int64(1), object(8)
memory usage: 12.4+ KB


Based on the Data classification , from the given data set we can divide 2 types and also by observing data we can say what kind the data type will fall

##### Categorical (Qualitative) Data:

Nominal: These are categories without any order. In the dataset, Surname, Name, Gender, Ethnicity, Department, and Position are nominal. They label categories but don't imply any rank.

Ordinal: These are categories with a specific order, but the difference between the values is not consistent. The dataset does not have a clear example of ordinal data unless there's a hidden order to positions that isn't explicitly stated.

##### Numerical (Quantitative) Data:

Discrete: These are countable values. Age is discrete since you count full years.

Continuous: These values can be measured and broken down into smaller parts, like fractions. Salary could be considered continuous as you can have fractions of a dollar.

Start_date is a bit special; it represents dates, which can be treated as ordinal because dates have a natural order. However, in raw form, it's often treated as an object or string in data frames. With proper processing, it can be converted to a date-time format for analysis.

### 2. Create a fictitious business that fits the data set

Let's create a fictitious business called "Future Techie Solutions.Inc" This business is a technology company providing software solutions to various industries.

"Future Techie Solutions.Inc" : is a technology company specializing in software solutions,the positions of Software Engineers, Network Engineers, and Database Administrators are crucial for technology company. With a diverse workforce, our company caters to various industries, offering tailored tech solutions for businesses ranging from healthcare to finance.

### 3. Develop a problem statement and 5 analytical questions to be investigated during your analysis.

##### Problem statement : 
As a Future Techie Solutions.Inc aims to maintain a diverse and inclusive workforce, we seek to understand the representation of different ethnicities across various departments and positions within the company.

Based on problem statement we came with these Analytical questions and  We aim to understand the distribution of ethnicities within different departments, analyze any salary disparities among employees of different ethnicities, and explore the relationship between ethnicity and other factors such as job positions and tenure.

##### Analytical questions

What is the percentage distribution of each ethnicity within different departments?                                     
Are there any significant differences in salary among employees of different ethnicities?                    
How does the average age of employees vary among different ethnic groups?                                      
How does the distribution of job positions vary among different ethnicities?                                              
Is there any disparity in the tenure of employees across various ethnic groups?

In this exercise, we're analyzing individual variables through univariate analysis. Among the provided analytical questions, the last three out of five are segmented univariate analysis. These questions delve into understanding the variations of variables like salary, age, job positions, and tenure across distinct segments such as departments, gender, and ethnicity.

### 4. Utilize descriptive statistics to analyze each of your analytical questions.

###### What is the percentage distribution of each ethnicity within different departments?

In [4]:
# Percentage distribution of each ethnicity within different departments
ethnicity_dept_percentage = df.groupby(['Department', 'Ethnicity']).size().unstack(fill_value=0)
ethnicity_dept_percentage = ethnicity_dept_percentage.div(ethnicity_dept_percentage.sum(axis=1), axis=0) * 100
print("Percentage distribution of each ethnicity within different departments:\n\n\n", ethnicity_dept_percentage)


Percentage distribution of each ethnicity within different departments:


 Ethnicity                 Asian  Black or African American  Hispanic  \
Department                                                             
Admin Offices         12.500000                  25.000000  0.000000   
Executive Office       0.000000                   0.000000  0.000000   
IT/IS                  7.692308                  11.538462  3.846154   
Production             9.433962                  20.754717  0.943396   
Sales                  3.846154                  30.769231  0.000000   
Software Engineering  14.285714                  14.285714  0.000000   

Ethnicity             Two or more races       White  
Department                                           
Admin Offices                  0.000000   62.500000  
Executive Office               0.000000  100.000000  
IT/IS                          3.846154   73.076923  
Production                     3.773585   65.094340  
Sales                   

##### Observations from the output
1. compared to all other types Hispanic is very low in stat as per company goal have to focus more to recruit talented employees over Hispanic ethnicity.
2. White are very more in all departments, try to implement new focused procedures to gradually reach towards businuess goal.
3. No department is near to ideal of our problem so we cannot take any department as refference for the progress of other departments.
4. frequent checks to be done in each quaters to check our reach to the businuess problem after implementing some measures.

##### Note

Descriptive Statistics Used:Frequency counts,percentages.Provides a breakdown of the distribution of ethnicities within departments, presenting the data in percentages to highlight the proportion of each ethnicity within each department.

while the analysis we may not directly involve traditional descriptive statistics like mean or standard deviation, it still relies on descriptive methods such as frequency counts and percentages to provide insights into the distribution of job positions across ethnic groups.

##### Are there any significant differences in salary among employees of different ethnicities?

In [5]:
salary_ethnicity_difference = df.groupby('Ethnicity')['Salary'].describe()
print("Salary differences among employees of different ethnicities:\n", salary_ethnicity_difference)


Salary differences among employees of different ethnicities:
                           count unique          top freq
Ethnicity                                               
Asian                        15     11   $45,760.00    3
Black or African American    36     23  $114,400.00    5
Hispanic                      2      2  $131,040.00    1
Two or more races             9      7  $114,400.00    3
White                       112     54  $114,400.00   11


##### Observations and Insights from the output

Ethnicity: This column lists different ethnic groups represented in the dataset.                      
count: Indicates the total number of entries (employees) for each ethnic group.                  
unique: Represents the number of unique salary values within each ethnic group.                          
top: Denotes the most frequently occurring salary within each ethnic group.                             
freq: Displays the frequency of occurrence of the most common salary within each ethnic group.

For employees identified as "Asian," there are 15 entries, with 11 unique salary values. The most common salary among Asian employees is $45,760.00, occurring 3 times within this group.
This analysis offers insights into the distribution of salaries among different ethnic groups within the company. It highlights the diversity of salaries within each ethnic group and may indicate potential disparities(Asians are facing disaparity here)

To mitigate this : patterns that warrant further investigation or consideration in HR policies and practices are required

##### Note
describe() method provides descriptive statistics for the salary distribution among different ethnicities. It includes statistics such as count, mean, standard deviation, minimum, 25th percentile (Q1), median (50th percentile), 75th percentile (Q3), and maximum salary.

##### How does the average age of employees vary among different ethnic groups?

In [6]:
# Average age variation among different ethnic groups
avg_age_ethnicity = df.groupby('Ethnicity')['Age'].mean()
print("Average age variation among different ethnic groups:\n", avg_age_ethnicity)


Average age variation among different ethnic groups:
 Ethnicity
Asian                        38.733333
Black or African American    36.805556
Hispanic                     33.500000
Two or more races            35.222222
White                        38.267857
Name: Age, dtype: float64


##### Observations and Insights from the output

1. Asians having the highest average age (38.73 years) and Hispanics having the lowest average age (33.50 years).                   
2. The higher average age among Asians and Whites could suggest a larger proportion of older employees within these ethnic groups, possibly indicating longer tenures or slower turnover rates.                                         
3. It may be beneficial for the company to analyze factors contributing to age disparities and ensure equitable opportunities and support for employees across all ethnicities and age groups.

##### Note
Descriptive Statistics Used: Mean. Calculates the mean age for each ethnicity, providing insights into the average age variation among different ethnic groups within the company.


##### How does the distribution of job positions vary among different ethnicities?

In [7]:
# Distribution of job positions among different ethnicities
position_ethnicity_distribution = df.groupby('Ethnicity')['Position'].value_counts(normalize=True).unstack(fill_value=0)
print("Distribution of job positions among different ethnicities:\n", position_ethnicity_distribution)

Distribution of job positions among different ethnicities:
 Position                   Accountant I  Administrative Assistant  \
Ethnicity                                                           
Asian                          0.000000                  0.000000   
Black or African American      0.055556                  0.000000   
Hispanic                       0.000000                  0.000000   
Two or more races              0.000000                  0.000000   
White                          0.008929                  0.017857   

Position                   Area Sales Manager       CIO  \
Ethnicity                                                 
Asian                                0.066667  0.000000   
Black or African American            0.194444  0.000000   
Hispanic                             0.000000  0.000000   
Two or more races                    0.444444  0.000000   
White                                0.098214  0.008929   

Position                   Database Admini

##### Observations and Insights from the output
1. Some job positions seem to be more prevalent among certain ethnic groups like Sr. Network Engineer, software engineering manager, Shared services manager. while others show a more balanced distribution across ethnicities like Production Technician I.
2. The distribution of job positions among different ethnicities provides insights into career paths and opportunities for advancement within the company.
3. Understanding these patterns can help identify potential barriers to career progression for employees from specific ethnic backgrounds and inform diversity and inclusion initiatives.
4. Addressing any imbalances in job position distribution can contribute to fostering a more inclusive and equitable workplace culture.

##### Note :
while the analysis for the distribution of job positions among different ethnicities may not directly involve traditional descriptive statistics like mean or standard deviation, it still relies on descriptive methods such as frequency counts and percentages to provide insights into the distribution of job positions across ethnic groups.

##### Is there any disparity in the tenure of employees across various ethnic groups?

In [8]:
# Disparity in tenure of employees across various ethnic groups
# Assuming tenure is the difference between start_date and current date
df['Start_date'] = pd.to_datetime(df['Start_date'])
df['Tenure'] = (pd.to_datetime('now') - df['Start_date']).dt.days
tenure_ethnicity_difference = df.groupby('Ethnicity')['Tenure'].describe()
print("Disparity in tenure of employees across various ethnic groups:\n", tenure_ethnicity_difference)


Disparity in tenure of employees across various ethnic groups:
                            count         mean         std     min      25%  \
Ethnicity                                                                    
Asian                       15.0  3969.000000  477.644219  3283.0  3577.00   
Black or African American   36.0  4235.444444  694.079429  3283.0  3678.50   
Hispanic                     2.0  4330.500000  129.400541  4239.0  4284.75   
Two or more races            9.0  4314.555556  556.715392  3465.0  3913.00   
White                      112.0  3891.196429  627.739113  2979.0  3423.00   

                              50%      75%     max  
Ethnicity                                           
Asian                      3913.0  4354.00  4823.0  
Black or African American  4095.0  4642.50  5985.0  
Hispanic                   4330.5  4376.25  4422.0  
Two or more races          4522.0  4767.00  4823.0  
White                      3731.0  4278.75  6650.0  


In [9]:
# Tenure values in days
tenure_in_days = {
    "Asian": 3969.0,
    "Black or African American": 4235.444444,
    "Hispanic": 4330.5,
    "Two or more races": 4314.555556,
    "White": 3891.196429
}

# Convert tenure from days to years
tenure_in_years = {ethnicity: tenure / 365.25 for ethnicity, tenure in tenure_in_days.items()}

# Print the tenure in years
for ethnicity, tenure in tenure_in_years.items():
    print(f"{ethnicity}: {tenure:.2f} years")


Asian: 10.87 years
Black or African American: 11.60 years
Hispanic: 11.86 years
Two or more races: 11.81 years
White: 10.65 years


##### Observations and Insights from the output
1. There is variation in average tenure among different ethnic groups, with Hispanic employees showing the highest average tenure (11.86 years), followed closely by Two or more races (11.81 years) and Black or African American employees (11.60 years) coming to Asian employees have the lowest average tenure (10.87 years) and White employees(10.65 years).
2. The differences in average tenure among ethnic groups may reflect underlying factors such as retention rates, career advancement opportunities, or organizational culture.
3. Higher average tenures among certain ethnic groups may indicate greater loyalty or satisfaction with the company, while lower average tenures could suggest turnover issues or challenges in employee retention.

##### Note
Descriptive Statistics Used: Mean, standard deviation, minimum, maximum, quartiles. Calculates descriptive statistics for tenure (length of employment) among different ethnicities, providing insights into potential differences in tenure distribution among ethnic groups.

### 5. Recap of all your insights.

1. ##### Percentage Distribution of Each Ethnicity within Different Departments:
   - Hispanic representation is notably lower compared to other ethnicities across all departments, indicating a need to focus on recruiting talented individuals from this demographic.
   - White employees are predominant in all departments, suggesting the implementation of targeted procedures to gradually diversify the workforce in line with business goals.
   - No department reflects an ideal ethnic distribution, necessitating regular assessments to gauge progress and the effectiveness of implemented measures.

2. ##### Significant Differences in Salary among Employees of Different Ethnicities:
   - Disparities in salary distribution among ethnic groups, such as Asians facing disparities, underscore the importance of scrutinizing these patterns for potential biases and addressing them through HR policies and practices.
 
3. ##### Variation in Average Age of Employees Among Different Ethnic Groups:
   - Asians have the highest average age, while Hispanics have the lowest, indicating potential differences in workforce demographics.
   - Analysis of age disparities can guide efforts to ensure equitable opportunities and support for employees across all ethnicities and age groups.

4. ##### Distribution of Job Positions Among Different Ethnicities:
   - Certain job positions are more prevalent among specific ethnic groups, while others show a more balanced distribution.
   - Understanding these patterns aids in identifying potential barriers to career progression for employees from specific ethnic backgrounds and informs diversity and inclusion initiatives.

5. ##### Disparity in Tenure of Employees Across Various Ethnic Groups:
   - Variation in average tenure suggests differences in retention rates, career advancement opportunities, or organizational culture among ethnic groups.
   - Addressing these disparities is vital for promoting diversity, equity, and inclusion within the organization and fostering employee satisfaction and loyalty.

By leveraging these insights, the company can make informed decisions to enhance diversity, equity, and inclusion efforts, optimize HR practices, and create a more supportive and equitable workplace environment for all employees, regardless of ethnicity.

### 6. Summarize your findings for our fictitious business.

Summary of Findings for Future Techie Solutions Inc:

1. **Ethnic Diversity in Departments:**
   - The company exhibits significant disparities in ethnic representation across departments, with Hispanic employees notably underrepresented compared to other ethnicities. 

2. **Salary Disparities and Equity:**
   - Analysis reveals disparities in salary distribution among ethnic groups, highlighting potential biases that need to be addressed through fair and equitable policies.

3. **Workforce Demographics and Age Variation:**
   - The workforce demographics vary across ethnic groups, with Asians having the highest average age and Hispanics the lowest. Efforts should be made to provide equal opportunities and support for career advancement across all age groups and ethnicities.

4. **Career Progression and Job Distribution:**
   - Certain job positions show a skewed distribution among ethnic groups, indicating potential barriers to career progression for employees from specific backgrounds. Initiatives aimed at promoting diversity.

5. **Retention and Loyalty:**
   - Variation in average tenure suggests differences in retention rates and organizational culture among ethnic groups. Efforts to promote diversity and inclusion should include measures to enhance employee satisfaction, loyalty, and long-term retention.

In conclusion, by addressing the identified disparities  from the analysis, Future Techie Solutions Inc. can foster a more inclusive workplace culture and justifies the business problem in a better way.